python/tests/test_topology.py

```diff
@@ -97,6 +98,615 @@ def generate_segments(n, sequence_length=100, seed=None):
     return segs
 
 
+def kc_distance_tree(tree1, tree2, topo_v_age=0):
+    """
+    Returns the Kendall-Colijn distance between the specified pair of trees.
+    topo_v_age determines weight of topology vs branch lengths in calculating
+    the distance. Set topo_v_age at 0 to only consider topology, set at 1 to
+    only consider branch lengths. See Kendall & Colijn (2016):
+    https://academic.oup.com/mbe/article/33/10/2735/2925548
+    """
+    samples = tree1.tree_sequence.samples()
+    if not np.array_equal(samples, tree2.tree_sequence.samples()):
+        raise ValueError("Trees must have the same samples")
+    if not len(tree1.roots) == len(tree2.roots) == 1:
+        raise ValueError("Trees must have one root")
+    k = samples.shape[0]
+    n = (k * (k - 1)) // 2
+    m = [np.ones(n + k), np.ones(n + k)]
+    M = [np.zeros(n + k), np.zeros(n + k)]
+    for tree_index, tree in enumerate([tree1, tree2]):
+        stack = [(tree.root, 0, tree.time(tree.root))]
+        while len(stack) > 0:
+            u, depth, time = stack.pop()
+            children = tree.children(u)
+            for v in children:
+                stack.append((v, depth + 1, tree.time(v)))
+            for c1, c2 in itertools.combinations(children, 2):
+                for u in tree.samples(c1):
+                    for v in tree.samples(c2):
+                        a = min(u, v)
+                        b = max(u, v)
+                        pair_index = a * (a - 2 * k + 1) // -2 + b - a - 1
+                        assert m[tree_index][pair_index] == 1
+                        m[tree_index][pair_index] = depth
+                        M[tree_index][pair_index] = tree.time(tree.root) - time
+            if len(tree.children(u)) == 0:
+                M[tree_index][u + n] = tree.branch_length(u)
+    return np.linalg.norm((1 - topo_v_age) * (m[0] - m[1]) + topo_v_age * (M[0] - M[1]))
+
+
+def kc_distance_tree_simple(tree1, tree2, topo_v_age=0):
```

Comment:
I don't like the topo_v_age parameter. I think it would be better to use the original terminology of calling it alpha or  beta or whatever, and we can explain what this means in the documentation.